src/NBitcoin/ProvenBlockHeader.cs

```diff
@@ -110,6 +113,8 @@ public override void ReadWrite(BitcoinStream stream)
 
             stream.ReadWrite(ref this.coinstake);
             this.CoinstakeSize = stream.ProcessedBytes - baseByteSize - this.MerkleProofSize - this.SignatureSize;
+
+            this.HeaderSize = baseByteSize;
```

Comment:
Why not get rid of `baseByteSize` and use this instead?

Reply:
ok